synthtool/protos/README.md

```diff
@@ -0,0 +1,27 @@
+# Synthtool and Autosynth Protocol
+
+**Synthtool** generates client library source code according to the currrent
+state of the source repo, and upstream repos like 
+[googleapis/googleapis](https://github.com/googleapis/googleapis).
+
+**Autosynth** periodically runs Synthtool for each client library source repo.
+Additionally, it repeatedly runs Synthtool in a binary search pattern to
+discover which upstream change triggered a change in the generate code.
+
+The proto files in this directory define the interface between.
+
+![Diagram of Synthtool and Autosynth](./diagram.png)
+
+**Synthtool** generates a file called `synth.metadata`, which contains a

```

Comment:
I don't know whether this is the right place to document it, but I had to go to some lengths to end up with the right information in synth.metadata, because Synthtool overwrites synth.metadata itself after synth.py completes. Some options:

- This is only an issue for C# due to the way that things work; don't bother documenting or changing
- Synthtool can be changed to automatically generate synth.metadata with the right information in, based on preclonedRepos, so I can delete some code - and we just need to document that here
- We change Synthtool in some other way that makes it easier
- We document the workaround here

Happy to chat about this tomorrow.

Reply:
I like the first bullet point.